webapp/src/components/ClassOverlapTable.test.tsx

```diff
@@ -0,0 +1,77 @@
+import { fireEvent, screen, waitFor, within } from "@testing-library/react";
+import { renderWithRouterAndRedux } from "mocks/utils";
+import ClassOverlapTable from "./ClassOverlapTable";
+import { AvailableDatasetSplits } from "types/api";
+import { setupServer } from "msw/node";
+import { getClassOverlapAPIResponse } from "mocks/api/mockClassOverlapAPI";
+import { QueryPipelineState } from "types/models";
+
+const renderClassOverlapTable = (
+  availableDatasetSplits: AvailableDatasetSplits,
+  pipeline: QueryPipelineState
+) =>
+  renderWithRouterAndRedux(
+    <ClassOverlapTable
+      jobId="local"
+      pipeline={pipeline}
+      availableDatasetSplits={availableDatasetSplits}
+    />,
+    { route: "/local?pipeline_index=0", path: "/:jobId?:pipeline" }
+  );
+
+describe("ClassOverlapTable", () => {
+  const handlers = [getClassOverlapAPIResponse];
+  const server = setupServer(...handlers);
+
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  afterAll(() => server.close());
+
+  it("should have expected columns", () => {
+    renderClassOverlapTable({ train: true, eval: true }, { pipelineIndex: 0 });
+    const expectedColumnHeaders = [
+      "Source Class",
+      "Target Class",
+      "Semantic Overlap Score",
+      "Pipeline Confusion",
+      "Utterances with Overlap",
+    ];
+    const columnHeaderDescription = [
+      "The class label of the samples being analyzed.",
+      "The class that the source class may look like. For pipeline confusion, this is the prediction.",
+      "Class overlap measures the extent to which source class samples are semantically similar to target class samples, in the training data. The score comes from both the proportion of samples and their degree of similarity.",
+      "Pipeline confusion indicates whether source class samples in the evaluation set are predicted to be in the target class.",
+      "Percent of source class samples that semantically overlap the target class (all in the training set).",
+    ];
+    const actualColumnHeaders = screen.getAllByRole("columnheader");
+    expectedColumnHeaders.forEach((name, index) => {
+      expect(actualColumnHeaders[index].textContent).toEqual(name);
+      within(actualColumnHeaders[index]).getByLabelText(
+        columnHeaderDescription[index]
+      );
+    });
+  });
+
+  it("should display the default sort as desc for colum header 'Semantic Overlap Score'", () => {
+    renderClassOverlapTable({ train: true, eval: true }, { pipelineIndex: 0 });
+    const header = screen.getByRole("columnheader", {
+      name: /Class overlap/,
+    });
```

Comment:
It is not obvious that the `name: /Class overlap/` corresponds to the `Semantic Overlap Score` column header, since it is the tooltip. Should we add something like `expect(header).toHaveTextContent("Semantic Overlap Score");`? Also, the `getByRole()` fits on one line:
```suggestion
    const header = screen.getByRole("columnheader", { name: /Class overlap/ });
    expect(header).toHaveTextContent("Semantic Overlap Score");
```

Reply:
Yeah, true; it needs to be more readable about which column we are targeting to test. Here, the purpose is to check the aria-sort of the column. The Name attribute has stored the tooltip, So I handled the test case using the aria label.

Reply:
Yes, that is fine. I'm just suggesting adding this simple assertion to make sure. Also, I find it makes the test more readable. Otherwise, when I first read the test, I thought it was a mistake.